activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -59,7 +59,8 @@ def create_join_table(table_1, table_2, **options)
         end
 
         def add_reference(table_name, ref_name, **options)
-          ReferenceDefinition.new(ref_name, **options).add_to(update_table_definition(table_name, self))
+          ReferenceDefinition.new(ref_name, **options)
+            .add_to(connection.update_table_definition(proper_table_name(table_name, table_name_options), self))
```

Comment:
Why do we need to call `proper_table_name`?

Reply:
I'm not completely sure if this is needed but I wanted to have maximum compatibility with existing code, especially with this line https://github.com/rails/rails/blob/master/activerecord/lib/active_record/migration.rb#L921

Reply:
Ah. We don't need that. Just calling `update_table_definition` on connection is enough.

Reply:
Updated.